runbotics-orchestrator/src/main/java/com/runbotics/service/impl/TenantServiceImpl.java

```diff
@@ -102,20 +121,26 @@ public Optional<TenantDTO> partialUpdate(TenantDTO tenantDTO) {
             throw new BadRequestAlertException("ID is not provided", ENTITY_NAME, "idNotProvided");
         }
 
+        tenantRepository.findByName(tenantDTO.getName()).ifPresent(tenant -> {
+            if (!tenant.getId().equals(tenantDTO.getId())) {
+                throw new BadRequestAlertException("Name is already used", ENTITY_NAME, "nameNotAvailable");
+            }
+        });
+
+        final User currentUser = userService.getUserWithAuthorities().get();
+
         return tenantRepository
             .findById(tenantDTO.getId())
-            .map(
-                tenant -> {
-                    if (tenantDTO.getCreatedById() != null) {
-                        final User updatedUser = userRepository
-                            .findById(tenantDTO.getCreatedById())
-                            .orElseThrow(
-                                () -> new BadRequestAlertException("Could not found user while updating", ENTITY_NAME, "userNotFound")
-                            );
-                        tenant.setCreatedBy(updatedUser);
-                    }
-                    tenantDTO.setUpdated(ZonedDateTime.now());
-                    tenantMapper.partialUpdate(tenant, tenantDTO);
+            .map(tenant -> {
+                if (tenantDTO.getCreatedBy() != null) {
+                    final User updatedUser = userRepository.findById(tenantDTO.getCreatedBy().getId()).orElseThrow(
+                        () -> new BadRequestAlertException("Could not found user while updating", ENTITY_NAME, "userNotFound")
```

Comment:
I haven't seen using "userNotFound" as translation key. Consider implementing this